src/core/pointcloud/qgspointcloudlayerrenderer.cpp

```diff
@@ -33,19 +33,26 @@
 #include "qgsmessagelog.h"
 #include "qgsmapclippingutils.h"
 #include "qgspointcloudblockrequest.h"
+#include "qgsruntimeprofiler.h"
+#include "qgsapplication.h"
 
 QgsPointCloudLayerRenderer::QgsPointCloudLayerRenderer( QgsPointCloudLayer *layer, QgsRenderContext &context )
   : QgsMapLayerRenderer( layer->id(), &context )
   , mLayer( layer )
+  , mLayerName( layer->name() )
```

Comment:
Why the need to add a mLayerName member when you store the layer pointer too? Are you hoping to get rid of the layer pointer in the near future?

Reply:
yeah -- it's not safe to use at all (outside of the constructor). I'm just minimising existing bad practice